### PR TITLE
feat: add Config.Wrap

### PR DESCRIPTION
**Pull request by tim-smart**

### CODE REVIEW

**Review by IMax153 — CHANGES_REQUESTED** [review]
A question - I haven't opened this in an editor, but I want to verify that the `Wrap` type will require properties to be an instance of `Config` (i.e primitive key/value pairs should not be allowed).

**Comment by tim-smart** [comment]
Yes all primitives are required to be `Config` instances. It only transforms the `struct`'s.

File: src/Config.ts [hunk 1]
```diff
@@ -57,6 +57,21 @@ export declare namespace Config {
   }
 }
 
+/**
+ * Wraps a type, turning it's members into `Config` entries.
```
[the remark this answers is not in the document]
Comment (IMax153) [reply src/Config.ts:61]: @tim-smart - would you be open to updating these docs?

File: src/Config.ts [hunk 2]
```diff
@@ -284,6 +299,28 @@ export const tuple: <T extends NonEmptyArrayConfig>(
   ...tuple: T
 ) => Config<TupleConfig<T>> = internal.tuple
 
+/**
+ * Constructs a config from a `Wrap`ed value.
+ *
+ * @since 1.0.0
+ * @category constructors
+ */
+export const unwrap = <A>(wrapped: Wrap<A>): Config<A> => {
```
Comment (IMax153) [inline src/Config.ts:308]: The implementation should live in the `internal` directory with the others, and this file should only re-export the method.

File: src/Config.ts [hunk 3]
```diff
@@ -284,6 +299,28 @@ export const tuple: <T extends NonEmptyArrayConfig>(
   ...tuple: T
 ) => Config<TupleConfig<T>> = internal.tuple
 
+/**
+ * Constructs a config from a `Wrap`ed value.
```
Comment (IMax153) [inline src/Config.ts:303]: I was hoping to add some additional documentation about how this method behaves here as well.

**Review by IMax153 — CHANGES_REQUESTED** [review]

**Review by IMax153 — APPROVED** [review]

**Comment by IMax153** [comment]
@tim-smart - given we've just merged `v0.1.0`, you may need to resolve some conflicts.

Also - you'll need to run `pnpm changeset` to generate a changeset for this PR.

Otherwise looks good to me! Thanks 👍

**Comment by IMax153** [comment]
One last thing we usually do for our PRs is rebase them down to a single commit. 

This way the `CHANGELOG` gets updated with a nice message and points to the single commit that we made the update in.

So once you're done editing, I'd suggest doing that so we can get this feature merged 🙂

**Comment by tim-smart** [comment]
> One last thing we usually do for our PRs is rebase them down to a single commit.
> 
> This way the `CHANGELOG` gets updated with a nice message and points to the single commit that we made the update in.
> 
> So once you're done editing, I'd suggest doing that so we can get this feature merged 🙂

👍🏻

**Pull request merged**